dora/core/common/src/main/java/alluxio/uri/UfsUrl.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.uri;
+
+import alluxio.AlluxioURI;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.InvalidPathException;
+import alluxio.grpc.UfsUrlMessage;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Preconditions;
+import org.apache.logging.log4j.util.Strings;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class UfsUrl {
+
+  public static final String SCHEME_SEPERATOR = "://";
+  public static final String PATH_SEPERATOR = "/";
+  UfsUrlMessage mProto;
+
+  public UfsUrl(UfsUrlMessage proto) {
+    Preconditions.checkArgument(proto.getPathComponentsList().size() != 0,
+        "The proto.path is empty, please check the proto first");
+    mProto = proto;
+  }
+
+  public UfsUrl(String ufsPath) {
+    // TODO(Tony Sun): Considering the case below:
+    //  when scheme does not exist, how to determine the scheme, or an empty scheme.
+    Preconditions.checkArgument(!ufsPath.isEmpty(),
+        "ufsPath is empty, please input a non-empty ufsPath.");
+    List<String> preprocessingPathList = Arrays.asList(ufsPath.split(SCHEME_SEPERATOR));
+    String scheme;
+    String authorityAndPath;
+    String rootDir = Configuration.getString(PropertyKey.DORA_CLIENT_UFS_ROOT);
+    Preconditions.checkArgument(preprocessingPathList.size() <= 2,
+        "There are multiple schemes, the input may contain more than one path, "
+            + "current version only support inputting one path each time.");
+    // Now, there are two condition. size = 1 -> without scheme; size = 2 -> with a scheme.
+    if (preprocessingPathList.size() == 1)  {
+      // If without scheme, set default scheme to local, i.e. "file".
+      String[] rootDirArray = rootDir.split(SCHEME_SEPERATOR);
+      if (rootDirArray.length == 1) {
+        scheme = "alluxio";
+      } else {
+        scheme = rootDirArray[0];
+      }
+      authorityAndPath = preprocessingPathList.get(0);
+    } else {
+      // preprocessingPathList.size() == 2, i.e. the ufsPath has one scheme.
+      scheme = preprocessingPathList.get(0);
+      authorityAndPath = preprocessingPathList.get(1);
+    }
+    Preconditions.checkNotNull(scheme, "scheme is empty, please input again");
+    Preconditions.checkNotNull(authorityAndPath, "authority or path is empty, please input again");
+    Preconditions.checkArgument(scheme.equalsIgnoreCase("file")
```

Comment:
i actually feel this check is not necessary. We do support other S3-like schemes like OSS, COS etc. Also there are in-house HDFS implementations which may use `hcfs://` or something else.

Reply:
I just use the check to emphasize we only support s3, local, hdfs in this PR.